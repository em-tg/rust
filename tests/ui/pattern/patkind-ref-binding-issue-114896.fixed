//@ check-pass
//@ run-rustfix
#![allow(dead_code)]

fn main() {
    fn x(a: &char) {
        let &(mut b) = a;
        b.make_ascii_uppercase();
//~^ WARNING cannot borrow `b` as mutable, as it is not declared as mutable
    }
}
