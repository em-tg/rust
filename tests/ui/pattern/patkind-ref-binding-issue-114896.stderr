warning[E0596]: cannot borrow `b` as mutable, as it is not declared as mutable
  --> $DIR/patkind-ref-binding-issue-114896.rs:8:9
   |
LL |         b.make_ascii_uppercase();
   |         ^ cannot borrow as mutable
   |
   = note: `#[warn(mut_non_mut)]` on by default
help: consider changing this to be mutable
   |
LL |         let &(mut b) = a;
   |             ~~~~~  +

warning: 1 warning emitted

For more information about this error, try `rustc --explain E0596`.
